roles/calico/README.rst

```diff
@@ -0,0 +1,147 @@
+Calico
+======
+
+.. versionadded:: 0.4
+
+`Calico <http://www.projectcalico.org>`_ is used in the project to add the IP
+per container functionality. Calico connects Docker containers through IP no matter
+which worker node they are on. Calico uses :doc:`etcd` to distribute information
+about workloads, endpoints, and policy to each worker node. Endpoints are
+network interfaces associated with workloads. Calico is deployed in the Docker
+container on each worker node and managed by systemd. Any workload managed by
+Calico is registered as a service in Consul.
+
+Modes
+^^^^^
+
+Calico can run a public cloud environment that does not allow either L3 peering
+or L2 connectivity between Calico hosts. Calico will then route traffic between
+the Calico hosts using IP in IP mode. At this time, the full node-to-node BGP
+mesh is supported and configured in CiscoCloud only. Other cloud environments
```

Comment:
We include [openstack.yml](https://github.com/CiscoCloud/microservices-infrastructure/blob/feature/calico-role/roles/calico/tasks/openstack.yml) with all necessary configuration. The idea is to automate configuration as much as possible, that's why there are no special instructions.